File: shake/src/main/java/com/example/shake/UploadCallback.java
package com.example.shake;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Response;

/**
 * @author fqxyi
 * @date 2017/4/11
 */
public interface UploadCallback {

    void onFailure(Call call, IOException e);

    void onResponse(Call call, Response response) throws IOException;

}

File: module/shake/src/main/java/com/example/shake/UploadCallback.kt
package com.example.shake

import okhttp3.Call
import okhttp3.Response
import java.io.IOException

/**
 * @author fqxyi
 * @date 2017/4/11
 */
interface UploadCallback {
    fun onFailure(call: Call, e: IOException)

    @Throws(IOException::class)
    fun onResponse(call: Call, response: Response)
}